### PR TITLE
Feature/custom sniff testing

### PR DESCRIPTION
**Pull request by essmahr**
This adds the boilerplate required to test any custom sniffs we might want to add.

- A new class that extends PHPUnit's `TestCase` to add a `getLinesForSniff`method
- an `AllSniffs.php` that gathers and runs tests in the project against fixture files
- Documentation and example 😃

### CODE REVIEW

File: PixelUnion/Tests/AllSniffs.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+            // Tests must have the extension 'php'.
+            $ext   = array_pop($parts);
+            if ($ext !== 'php') {
+                continue;
+            }
+
+            // Tests must end in 'Test'.
+            $fileName = array_pop($parts);
+            if (substr_compare($fileName, 'Test', -strlen('Test'))) {
```
Comment (mojitoj) [inline PixelUnion/Tests/AllSniffs.php]: - This seems to need a `!`. 
- Edge cases for files with no extensions or file names starting with `.` will cause exception.

File: PixelUnion/Tests/AllSniffs.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    public static function suite()
+    {
+        $suite = new TestSuite('Pixel Union CodeSniffing Standard');
+
+        $testsDir = __DIR__;
+
+        $di = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($testsDir));
+        foreach ($di as $file) {
+            // Tests must end in *Test.php
+            $fileEndCondition = 'Test.php';
```
Comment (mojitoj) [inline PixelUnion/Tests/AllSniffs.php]: This doesn't need to be inside the loop.

**Review by mojitoj — APPROVED** [review]

**Review by matthewlehner — APPROVED** [review]
I think this is good!

It looks good!

There was something about ternaries in a past PR, but I can't remember what that was now.
